File: C#/ToyLanguage/ToyLanguage/Model/Collections/WrapperList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguage.Exceptions;
using ToyLanguage.Interfaces;

namespace ToyLanguage.Model.Collections
{
    class WrapperList<T> : IMyList<T>
    {
        private List<T> data;
            
        public WrapperList()
        {
            data = new List<T>();
        }

        public void add(T element)
        {
            data.Add(element);
        }

        public T get(int position)
        {
            if (position < 0 || position >= data.Count)
            {
                throw new InvalidPositionException();
            }
            return data.ElementAt(position);
        }

        public bool isEmpty()
        {
            if(data.Count == 0)
            {
                return true;
            }
            return false;
        }

        public void remove(int position)
        {
            if(position < 0 || position >= data.Count)
            {
                throw new InvalidPositionException();
            }
            data.RemoveAt(position);
        }

        public int size()
        {
            return data.Count;
        }

        public string ToString()
        {
            StringBuilder result = new StringBuilder();
            foreach (T element in data){
                result.Append(element.ToString() + "\n");
            }
            return result.ToString();
        }
    }
}

File: Java/src/model/Collections/WrapperList.java
package model.Collections;

import interfaces.IList;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Lucian on 11/2/2015.
 */
public class WrapperList<T> implements IList<T>{

    /**
     * The storage list
     */
    private List<T> data;

    /**
     * The constructor
     */
    public WrapperList(){
        data = new ArrayList<>();
    }

    /**
     * Add an element in list
     *
     * @param element T element
     */
    @Override
    public void add(T element) {
        data.add(element);
    }

    /**
     * Get element from list by position
     *
     * @param position Int position
     * @return The element
     */
    @Override
    public T get(int position) {
        return data.get(position);
    }

    /**
     * Remove element from list by position
     *
     * @param position Int position
     */
    @Override
    public void remove(int position) {
        data.remove(position);
    }

    /**
     * Check if list is empty
     *
     * @return True if yes, false otherwise
     */
    @Override
    public boolean isEmpty() {
        return data.isEmpty();
    }

    /**
     * Get the size of the list
     *
     * @return int size
     */
    @Override
    public int size() {
        return data.size();
    }

    /**
     * String representation
     *
     * @return String
     */
    @Override
    public String toString() {
        StringBuilder result = new StringBuilder();
        for(T element : data){
            result.append(element.toString() + "\n");
        }
        return result.toString();
    }
}
